Woof/Woof/Model/Preferences/PreferencesHandler.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+
+/**
+ A type used for managing user preferences in the application.
+ */
+enum PreferencesHandler {
+    // MARK: - Internal interface
+
+    /**
+     Saves the selected role as a user preference.
+
+     - Parameters:
+        - userRole: The selected role to be saved.
+
+     - Returns: A boolean value indicating whether the saving operation was successful.
+     */
+    @discardableResult static func set(userRole: Role) -> Bool {
+        var preferences: Preferences = loadPreferences() ?? Preferences()
+
+        preferences.selectedRole = userRole
+
+        return store(preferences)
+    }
+
+    /**
+     Loads the selected role from user preferences.
+
+     - Returns: The selected role for the user in the app session. If the role couldn't be loaded, returns `.none`.
```

Comment:
> If the role couldn't be loaded, returns `.none`.

So the `.none` value will be returned if and only if the role couldn't be loaded (some error occurred as far as I understood), right?

Reply:
No, except for a load error, there may still be an option when the settings are missing in the storage

Reply:
fixed